elvish-mode.el

```diff
@@ -63,7 +63,7 @@ expected.")
 
 (defconst elvish-keyword-pattern
   (let ((keywords (cons 'or elvish-keywords)))
-    (eval `(rx (group ,keywords) (not word))))
+    (eval `(rx (not (any word ?_ ?-)) (group ,keywords) (not (any word ?_ ?-)))))
```

Comment:
Nice catch, butI think I'd prefer to do something like `(defconst not-elvish-symbol-component '(and (not-syntax word) (not-syntax symbol)))` and use that in place of the `(not (any ...`